python/cudf/cudf/core/reshape.py

```diff
@@ -449,11 +470,16 @@ def length_check(obj, name):
     else:
         result_df = df.drop(labels=columns)
         for name in columns:
-            if hasattr(df[name]._column, "categories"):
+            if isinstance(df[name]._column, CategoricalColumn):
                 unique = df[name]._column.categories
             else:
                 unique = df[name].unique()
 
+            if dummy_na is False:
```

Comment:
`0 is False` evaluates to `False`:
```suggestion
            if not dummy_na:
```